### ISSUE TITLE
remesh_botsch does not work for non-closed meshes

### ISSUE THREAD

**Issue by EmJay276** [sgsellan/gpytoolbox#30]
I tried to remesh a non-closed/ watertight mesh with the remesh_botsch algorithm.
Compared to a closed mesh (with almost the same size) it takes a lot longer (>100 times) and returns an error by exiting my python console.

- Is your implementation not useable for non-closed meshes? The original paper has options for the boundary.

**Comment by sgsellan** [comment]
Hi EmJay! If I recall correctly, you must fix the boundary vertices as feature vertices for the method to work. Actually remising the boundary is not supported. Could you please attach the mesh you used and the exact command you ran to encounter this problem?

**Comment by EmJay276** [comment]
I will try to build a minimal example, my code / mesh is currently rather complicated (result of an toplogy optimization, I want to remesh partially).

**Comment by sgsellan** [comment]
Hi EmJay!

The PR #32 should fix this issue. Running your command from above now works without issue (the boundary vertices are not remeshed). Once the PR is merged, you can either wait for the 0.0.3 release or build the library locally running `pip install .` Thanks for raising this issue and for using our library!

-Silvia

**Comment by EmJay276** [comment]
Hi Silvia,
thank you very much for the very fast fix and also adding the feature vector, that's exactly what I needed!
Michael

**Comment by EmJay276** [comment]
@sgsellan 
Hi Silvia,
is it somehow possible to keep track of the "feature" vector or at least of the points which are removed?

Before remeshing I know which point is is on which face of my model (I need this info for edge detection / projection on the original surface - like a sphere / cylinder). But even the feature "points" change their index during remeshing.

Simplified example:
- Input:
  - Points - P0 = [A, B, C, D, E]
  - Feature points - F0 = [A, B, E] (indices 0, 1, 4 in P0)
- Remeshing removes point D
- Output points - P1 = [A, B, C*, E]
- Feature points now have the indices 0, 1, 3

I could try to find the index of each point of P1 in P0, but this seems very tedious and not very practical for large models.

-Michael

**Comment by EmJay276** [comment]
@sgsellan 
My current workaround for this is too reorder the input, so all fixed nodes are at the beginning of the array.

A solution would be to also return the indices of the fixed nodes after the remesh.

**Comment by sgsellan** [comment]
Interesting workaround! I'll set a low priority to this but it's a nice / feasible thing to add to the remember (would also be happy to merge a PR that did it 😉 !)

**Comment by EmJay276** [comment]
I can try to create a PR, but I'm not very familiar with C code. Don't count on me 😅

**Comment by sgsellan** [comment]
Oh, maybe even just a PR that reorders the mesh such that feature vertices are first and calls the C++ remesher like in your hotfix? We could just add in the documentation that the n fixed vertices are returned first. After all, the output ordering right now is arbitrary.

**Comment by EmJay276** [comment]
👍 I try to send the PR by end of the week
